vitrina/datasets/views.py

```diff
@@ -42,3 +47,50 @@ def get_context_data(self, **kwargs):
         }
         context_data.update(extra_context_data)
         return context_data
+
+
+class DatasetCreateView(CreateView):
+    model = Dataset
+    template_name = 'base_form.html'
+    context_object_name = 'dataset'
+    form_class = NewDatasetForm
+
+    def get(self, request, *args, **kwargs):
+        if request.user.is_authenticated and\
+                request.user.organization is not None and\
+                request.user.organization.slug == self.kwargs['slug']:
+            return super(DatasetCreateView, self).get(request, *args, **kwargs)
+        else:
+            return redirect('organization-detail', self.kwargs['org_kind'], self.kwargs['slug'])
+
+    def form_valid(self, form):
+        object = form.save(commit=False)
+        object.slug = slugify(object.title)
+        object.save()
+        return super(DatasetCreateView, self).form_valid(form)
+
+    def get_success_url(self):
+        return reverse_lazy('dataset-detail', kwargs={'slug': self.object.slug})
+
+
+class DatasetUpdateView(UpdateView):
+    model = Dataset
+    template_name = 'base_form.html'
+    context_object_name = 'dataset'
+    form_class = NewDatasetForm
+
+    def get(self, request, *args, **kwargs):
+        if request.user.is_authenticated and Dataset.objects.filter(manager=self.request.user).exists():
```

Comment:
You also need to check this for `post`, so better use `LoginRequiredMixin` and `PermissionRequiredMixin`.

https://docs.djangoproject.com/en/3.2/topics/auth/default/#the-permissionrequiredmixin-mixin